### PR TITLE
1388/updates to date component (breaking change)

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue

Addresses #1388
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

The only date field we had in the app was `DOBField` - but we need multiple date fields in Listings Management for things like application due date, postmark due date, event dates, etc. This ticket pulls any birthdate logic out of being default in the component into a new prop on the component (validateAge18) and renames the field to just be DateField. Confirmed with Jesse the places where we do wanted to be validating 18 and over and the places where we don't (validating for account creation and primary applicant in both public and partners, but not for household members).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

You can check out the stories and the places in the app where we use a birthdate field, which include account creation, account editing, primary applicant during application process, household applicant during application process. We don't currently have a vanilla date field but we will in the next few PRs as we build the Listings Management form out.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Comment by emilyjablonski** [comment]
Still working out the Cypress issues, but open to feedback about the approach still!

File: ui-components/src/forms/DateField.tsx [hunk 1]
```diff
@@ -4,49 +4,37 @@ import { Field } from "./Field"
...
   readerOnly?: boolean
+  validateAge18?: boolean
 }
 
-const DOBField = (props: DOBFieldProps) => {
-  const { defaultDOB, error, register, watch, atAge, validateHhAge, name, id, errorMessage } = props
+const DateField = (props: DateFieldProps) => {
+  const { defaultDate, error, register, watch, name, id, errorMessage } = props
   const fieldName = (baseName: string) => {
     return [name, baseName].filter((item) => item).join(".")
```
Comment (jaredcwhite) [inline ui-components/src/forms/DateField.tsx:30]: Hey I know this isn't part of the code you touched, but initially the term `fieldName` threw me (wasn't sure what it's for)…and even looking at the code now it's hard for me to initially grok why it's there and what it's doing. Any thoughts on refactoring and/or adding some comments?

**Review by jaredcwhite — COMMENTED** [review]
This approach looks good to me @emilyjablonski. Just noticed one function that threw me for a loop at first.

**Comment by jaredcwhite** [comment]
Just wondering…should we do some kind of future date validation? Like I can punch in year 5050 or whatever. I'm guessing if somebody did accidentally type 3031 instead of 2021 they'd notice that, but who knows…

File: ui-components/src/forms/DateField.tsx (hunk 1 above)
Comment (emilyjablonski) [reply ui-components/src/forms/DateField.tsx:30]: Sure!

**Comment by emilyjablonski** [comment]
Happy to add that in :)

**Comment by emilyjablonski** [comment]
Oops! Closing to re-open to the dev branch in a few

**Pull request closed without merging**